source/simulator/parameters.cc

```diff
@@ -367,6 +367,24 @@ namespace aspect
                            "block preconditioner for the Stokes equation can be found in "
                            "\\cite{KHB12}.");
 
+        prm.declare_entry ("Use full A block as preconditioner", "false",
+                           Patterns::Bool(),
+                           "This parameter determines whether we use an simplified approximation of the"
+                           "$A$ block as preconditioner for the Stokes solver, or the full $A$ block. The "
+                           "simplified approximation only contains the terms that describe the coupling of"
+                           "identical components (plus boundary conditions) as described in "
+                           "\\cite{KHB12}. The full block is closer to the description in "
+                           "\\cite{rudi2017weighted}. There is no clear way to determine, which preconditioner "
+                           "performs better in runtime, the default value (simplified approximation) requires "
+                           "more GMRES iterations, but is faster to apply in each iteration. On the downside it "
+                           "requires storing an additional matrix block, and such requires more memory, although "
+                           "the block is by a factor of 2 or 3 smaller than the original A block (depending on spatial "
```

Comment:
one of the reasons to use the simplified A is that AMG needs a lot less memory, so this is less clear to me. Also: right now we are not saving memory (but I could do that soon as a follow-up).